src/Enhavo/Bundle/ArticleBundle/Repository/ArticleRepository.php

```diff
@@ -84,4 +84,79 @@ public function findPublished()
 
         return $articles;
     }
+
+    public function findNextInDateOrder(Article $currentArticle)
+    {
+        // Find articles with same date
+        $query = $this->createQueryBuilder('n');
+        $query->andWhere('n.public = true');
+        $query->andWhere('n.publication_date <= :currentDate');
+        $query->andWhere('n.publication_date = :articleDate');
+        $query->setParameter('currentDate', new \DateTime());
```

Comment:
The filters for "public=true" and "publication date<=current date" are there for the list to be consistent with the other functions (findPublished, filterByDate, getMonths). These all have the same filters, and none of them can set the current date parameter. I recommend keeping it that way or changing it for all of these functions.